models/UserGroupRelation.ts

```diff
@@ -0,0 +1,29 @@
+import { Schema, model, models } from 'mongoose';
+
+const UserGroupRelationSchema = new Schema({
```

Comment:
You are using normalization in MongoDB, and it is justified. Your relationship schema looks good so far